core/templates/pages/exploration-player-page/services/player-transcript.service.ts

```diff
@@ -53,13 +53,11 @@ export class PlayerTranscriptService {
   }
 
   hasEncounteredStateBefore(stateName: string): boolean {
-    // TODO(#7165): Replace 'any' with the exact type.
-    return this.transcript.some((transcriptItem: any) => {
+    return this.transcript.some(transcriptItem => {
```

Comment:
type is not required for transcriptItem?

Reply:
the type is automatically inferred because the type of `this.transcript` is known.

Reply:
cool, thanks for explaining!